rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search/540_ignore_above_synthetic_source.yml

```diff
@@ -0,0 +1,179 @@
+---
+ignore_above mapping level setting:
+  - requires:
+      cluster_features: [ "mapper.ignore_above_index_level_setting" ]
+      reason: introduce ignore_above index level setting
+  - do:
+      indices.create:
+        index:  test
+        body:
+          settings:
+            index:
+              mapping:
+                ignore_above: 10
+          mappings:
+            _source:
+              mode: synthetic
+            properties:
+              keyword:
+                type: keyword
+              flattened:
+                type: flattened
+
+  - do:
+      index:
+        index:  test
+        refresh: true
+        id: "1"
+        body:   { "keyword": "foo bar", "flattened": { "value": "the quick brown fox" } }
+
+  - do:
+      search:
+        body:
+          fields:
+            - keyword
+            - flattened
+          query:
+            match_all: {}
+
+  - length: { hits.hits: 1 }
+  - match: { hits.hits.0._source.keyword: "foo bar" }
+  - match: { hits.hits.0._source.flattened.value: "the quick brown fox" }
+  - match: { hits.hits.0.fields.keyword.0: "foo bar" }
+
+---
+ignore_above mapping level setting on arrays:
+  - requires:
+      cluster_features: [ "mapper.ignore_above_index_level_setting" ]
+      reason: introduce ignore_above index level setting
+  - do:
+      indices.create:
+        index:  test
+        body:
+          settings:
+            index:
+              mapping:
+                ignore_above: 10
+          mappings:
+            _source:
+              mode: synthetic
+            properties:
+              keyword:
+                type: keyword
+              flattened:
+                type: flattened
+
+  - do:
+      index:
+        index:  test
+        refresh: true
+        id: "1"
+        body:   { "keyword": ["foo bar", "the quick brown fox"], "flattened": { "value": ["the quick brown fox", "jumps over"] } }
+
+  - do:
+      search:
+        body:
+          fields:
+            - keyword
+            - flattened
+          query:
+            match_all: {}
+
+  - length: { hits.hits: 1 }
+#TODO: synthetic source field reconstruction bug (see https://github.com/elastic/elasticsearch/issues/112044)
+#  - match: { hits.hits.0._source.keyword: ["foo bar", "the quick brown fox"] }
```

Comment:
The bug is for flattened but keyword is commented out?

Reply:
No the bug is for `keyword` I just forgot to create an issue similar to the flattened one. I put the link in the comment for my own reference since I believe it might be the same issue affecting both `keyword` and `flattened`). I would like to create a reproduction which uses the mapping level `ignore_above` instead of the setting level `ignore_above`.

Reply:
I am also a bit surprised that I don't see it for the flattened field too...but I didn't investigate that.

Reply:
I am going to remove the link and try to reproduce the issue with another test and link it to the flattened one.

Reply:
At the moment I am not able to reproduce the issue with `keyword` with just a `keyword` field. This makes me believe the actual issue is with flattened fields.